_posts/2021-08-09-image-resizing.md

```diff
@@ -0,0 +1,172 @@
+
+Other interesting results can be achieved using a pattern similar to the previous grid but with a border enlarged up to 22 pixels and with 20x20 white squares (note that we use a slightly different value for the border and the squares).
+
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="comparison on grid" src="/images/image-resizing/grid2_comparison.png" >
+        <figcaption>
+        Figure 3. Comparison of Pillow and OpenCV on a grid image.
+        From left to right, raw image (652x652), Pillow resized image with the nearest filter, OpenCV resized image with the nearest filter, Pillow resized image with the bilinear filter and OpenCV resized image with the bilinear filter.  
+        All the low-resolution images are 64x64 and are displayed with the same size as the raw image only for a better view.
+        </figcaption>
+    </figure>
+</div>
+
+With this input, in the resulting images, the number of squares is unchanged but in some cases, some rows or columns result having a rectangular shape.
+
+### Natural images
+
+We chose to use synthetic images because we want to bring to evidence the creation of artifacts and see the differences between the libraries. These differences are more difficult to see on natural images, i.e. the ones acquired with a camera. In this case, as you see can in the figures below, the results with the _nearest neighbor_ filter are quite the same while, using the _bilinear_ one the effect of the antialiasing filter become more visible.
+
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="nearest neighbor on dog" src="/images/image-resizing/dog_nearest.png" >
+        <figcaption>Figure 4. Nearest neighbor interpolation on a natural image.</figcaption>
+    </figure>
+</div>
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="bilinear on dog" src="/images/image-resizing/dog_bilinear.png" >
+        <figcaption>Figure 5. Bilinear interpolation on a natural image.</figcaption>
+    </figure>
+</div>
+
+<br/>
+
+## Conclusion
+
+In this post, we saw that even if image resizing is one of the most used image processing operations, could hide some traps and it is important to carefully choose the library we want to use to resize in particular, if you want to deploy your ML solution. 
+
+Since we noted that the most correct behavior is given by the `Pillow` resize and we are interested in deploying our applications in C++, it could be useful to use it in C++. 
+The `Pillow` [image processing](https://github.com/python-pillow/Pillow/blob/master/src/libImaging/) algorithms are almost all written in C, but they cannot be directly used because they are designed to be part of the python wrapper.  
```

Comment:
```suggestion
The `Pillow` [image processing](https://github.com/python-pillow/Pillow/blob/master/src/libImaging/) algorithms are almost all written in C, but they cannot be directly used because they are designed to be part of the Python wrapper.  
```